packages/pigeon/lib/swift_generator.dart

```diff
@@ -607,6 +580,63 @@ import FlutterMacOS
     indent.newln();
   }
 
+  void _writeDecodeCasting(
```

Comment:
this function is a bit complicated (looks like it's moving code). maybe add a comment explaining a bit?

Reply:
will do!